Secret-agency-web/src/main/react/secret-agency/src/App.tsx

```diff
@@ -18,85 +27,88 @@ type IState = IAppState;
-          this.setState((prevState) => ({...prevState, loginError: response}));
-        } else {
-          // put authenticated user to state and local storage (to enable automatic authentication)
-          this.setState((prevState) => ({...prevState, authenticatedAgent: response}));
-          secretAgencyRepository.storeAuthenticatedAgent(response);
-        }
-      })
-  };
+    /********************************************************
+     * EVENT HANDLERS
+     *******************************************************/
+    /**
+     * Authentication method. Try to authenticate agent with given credentials.
+     * @param codeName codename
+     * @param password password
+     */
+    private onLogin = (codeName: string, password: string): void => {
+        authenticationService.authenticate(codeName, password)
+            .then((response) => {
+                // error was returned
+                if (typeof response === "string") {
+                    this.setState((prevState) => ({...prevState, loginError: response}));
+                } else {
+                    // put authenticated user to state and local storage (to enable automatic authentication)
+                    this.setState((prevState) => ({...prevState, authenticatedAgent: response}));
+                    secretAgencyRepository.storeAuthenticatedAgent(response);
+                }
+            })
+    };
 
-  private onLogout = (e: MouseEvent<HTMLButtonElement>): void => {
-    e.preventDefault();
-    authenticationService.logOut();
-    this.setState((prevState) => ({...prevState, authenticatedAgent: null}));
-    secretAgencyRepository.removeAuthenticatedAgent();
-  };
+    private onLogout = (e: MouseEvent<HTMLButtonElement>): void => {
+        e.preventDefault();
+        authenticationService.logOut();
+        this.setState((prevState) => ({...prevState, authenticatedAgent: null}));
+        secretAgencyRepository.removeAuthenticatedAgent();
+    };
 
-  /**
-   * Hide errors when user starts to type.
-   */
-  private hideAuthenticationError = (): void => {
-    this.setState((prevState) => ({...prevState, loginError: ""}))
-  };
+    /**
+     * Hide errors when user starts to type.
+     */
+    private hideAuthenticationError = (): void => {
+        this.setState((prevState) => ({...prevState, loginError: ""}))
+    };
 
-  /********************************************************
-   * LIFE-CYCLE METHODS
-   *******************************************************/
-  constructor(props: {}) {
-    super(props);
+    /********************************************************
+     * LIFE-CYCLE METHODS
+     *******************************************************/
+    constructor(props: {}) {
+        super(props);
 
-    this.state = {
-      authenticatedAgent: null,
-      loginError: "",
+        this.state = {
+            authenticatedAgent: null,
+            loginError: "",
+            tabs: [
+                {title: "Agents", link: "agents"},
```

Comment:
Tabs jsou ve state proto, že se budou měnit na základě toho, jestli je přihlášený Admin, nebo User?

Reply:
Bol taký plán že to bude konfigurovatelné, keďtak sa potom upraví.

Reply:
Ok